### ISSUE TITLE
multiSight

### ISSUE THREAD

**Issue by Fjeanneret** [Bioconductor/Contributions#1902]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/Fjeanneret/multiSight

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.
- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/

**Comment by lshep** [comment]
Please ignore the previous comment concerning the issue being previously posted.  I was running updates and inadvertently caused this message

**Comment by Fjeanneret** [comment]
Understood, thank you!

**Comment by hpages** [comment]
Hi @Fjeanneret,

Thanks for submitting the multiSight package.

The vignette requires a little bit more attention. For example the "Installation" section is empty, `>` seems to be used inappropriately (it's usually used for quoting), "Why to use MultiSight ?" should be "Why use MultiSight ?", the name of the package is "multiSight" not "MultiSight", the "What is it ?" subsection in the "Why to use MultiSight ?" section (does it actually answer the "why" question, and if so then why is the subsection title a "what"?), the space before the question marks (yes in French but not in English), etc...

It might actually be easier for you to spot these problems by looking at the HTML vignette, which you can display with:
```
library(remotes)
install_git("https://git.bioconductor.org/packages/multiSight", build_vignettes=TRUE, force=TRUE)
library(multiSight)
browseVignettes("multiSight")
```

Thanks,
H.

**Comment by Fjeanneret** [comment]
Hi @hpages,

Thanks for your review and comments!

- I used ">" to highlight some points, therefore ">" have been replaced by italic style. 
- Installation section have been completed.
- Packages and method names have been normalized.
- Some typos have been withdrawn.

With my best wishes,
Florian

**Comment by hpages** [comment]
Hi Jean,

Thanks for the improvements. Unfortunately the vignette needs more work. The language is still too hard to follow. For example the very first sentence is:

> This vignette is built to be efficient as quickly as possible with multiSight.

I guess you mean something like "The purpose of this vignette is to help you become productive as quickly as possible with the multiSight package".

Then:

> multiSight is a R package providing an user-friendly graphical interface to analyze and explore your omic data sets in a multi-omic manner by DESeq2 (see Biological Insights tab), machine learning methods with biosigner and multi-block statistical analysis (see Classification tab) helped by p-values pooling Stouffer’s method.

I don't understand how the second part of the sentence (the part after the comma) is connected to the first part.

Even the first part is not clear. What does it mean to "explore omic data in a multi-omic manner by DESeq2"?

And so on...

Also please use an article where appropriate. For example instead of:

> ... to provide large panel of ...

it should be "to provide **a** large panel of".

I'm not a native English speaker myself and I understand the challenges of writing good technical documentation in English. Doesn't have to be perfect but people will need to be able to understand your vignette if they're going to use your package. Maybe seek help from an native English speaker? Have them read your vignette and provide some feedback?

Thanks for your efforts.

Best,
H.

**Comment by Fjeanneret** [comment]
Hi @hpages,

Thank you for your time and your comments.
Of course I fully agree, we will work on another version of this vignette to make it more understandable.

I will tag you when I finish.

Best regards,
Florian

**Comment by Fjeanneret** [comment]
Hi @hpages,

I propose a new version of our vignette. 
I hope this version is much better than the previous one.
Otherwise, I will do what is required.

Best regards, 
Florian

**Comment by hpages** [comment]
The vignette is better. Thank you for your efforts on this.

There is one last thing but it's an important one, and I'm sorry that I didn't catch this earlier: the package has 204 direct and indirect dependencies! This is huge. Having so many dependencies tends to make software very fragile. Please try to reduce the number of dependencies as much as you can.

Thanks again,
H.

**Comment by Fjeanneret** [comment]
Hi @hpages,

> Please try to reduce the number of dependencies as much as you can.

Understood, I deleted _ggplot2_(148 dependencies), _tidyr_(63), _purrr_(55) and _htmlwidgets_(44). 
I used some R base functions to replace them or just deleted minor functions.
I hope it is better now. 

Nevertheless, the graphical part of this package leads to import some big packages such as _Shiny_, _ShinyDashboard_, _R6_, and _igraph_.
More, the analysis core is based on packages like _mixOmics_, _biosigner_, _DESeq2_, _clusterProfiler_, _rWikiPathways_, _ReactomePA_ and _enrichplot_.
In fact, I think it seems complicated to remove one of these packages without removing a major function but maybe you have some suggestions about this?

Thank you for your time,

Best,
Florian

**Comment by hpages** [comment]
Hi @Fjeanneret,

> Understood, I deleted ggplot2(148 dependencies), tidyr(63), purrr(55) and htmlwidgets(44).

I'm not sure I understand these numbers. Are you saying that **ggplot2** has 148 dependencies? `sessionInfo()` only reports 26 dependencies for **ggplot2**.

Note that even though you've removed **ggplot2** from your direct deps, your package still depends on it via other deps so **ggplot2** still gets loaded when I do `library(multiSight)`, as indicated by `sessionInfo()`. So in the end this removal didn't change anything. Same for **tidyr**, **purrr**, and **htmlwidgets**.

It seems to me that you are abusing the `@import` directive by importing things that you only need in your examples or vignette (e.g. **org.Mm.eg.db**). These things should be in the `Suggests` field, not in the `Imports` field.

Thanks,
H.

**Comment by Fjeanneret** [comment]
> Hi @Fjeanneret,
> 
> > Understood, I deleted ggplot2(148 dependencies), tidyr(63), purrr(55) and htmlwidgets(44).
> 
> I'm not sure I understand these numbers. Are you saying that **ggplot2** has 148 dependencies? `sessionInfo()` only reports 26 dependencies for **ggplot2**.
> 
> Note that even though you've removed **ggplot2** from your direct deps, your package still depends on it via other deps so **ggplot2** still gets loaded when I do `library(multiSight)`, as indicated by `sessionInfo()`. So in the end this removal didn't change anything. Same for **tidyr**, **purrr**, and **htmlwidgets**.
> 
> It seems to me that you are abusing the `@import` directive by importing things that you only need in your examples or vignette (e.g. **org.Mm.eg.db**). These things should be in the `Suggests` field, not in the `Imports` field.
> 
> Thanks,
> H.

Hi @hpages, 

Yes thank you I hadn't notice this, in fact I've move **org.Mm.eg.db** to Suggests field.  I hope it's better now. Actually, I don't see other package in this case. The other packages are used by package functions.

In fact, many packages are imported to put them together in a useful GUI for non-programmers.

Secondly, I have a Warning from the Windows server I can not reproduce: 
_" Warning: no function found corresponding to methods exports from 'S4Vectors' for: 'combineRows' "_
Do you know whether is it due to recent changes in S4Vectors?
Of course, for more questions about this issue I will move to the specific slack channel.

Thanks for your time,
Florian

**Comment by hpages** [comment]
Thanks for trying to reduce the number of dependencies. It's not always easy but it's important to keep in mind that the less dependencies the better.

The warning about combineRows on the Windows server was a problem on our side. It has been cleared now.

Thanks for your contribution.

H.

**Comment by Fjeanneret** [comment]
Thank you for your very helpful comments @hpages. 

See you next time!

**Comment by vjcitn** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/Fjeanneret.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("multiSight")`. The package 'landing page' will be created at

https://bioconductor.org/packages/multiSight

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
